Sources/Basic/JSON.swift

```diff
@@ -122,23 +130,22 @@ extension JSON: ByteStreamable {
         case .string(let value):
             stream <<< Format.asJSON(value)
         case .array(let contents):
-            // FIXME: OutputByteStream should just let us do this via conformances.
-            stream <<< "["
+            stream <<< "[" <<< (shouldIndent ? "\n" : "")
             for (i, item) in contents.enumerated() {
-                if i != 0 { stream <<< ", " }
-                stream <<< item
+                if i != 0 { stream <<< "," <<< (shouldIndent ? "\n" : " ") }
+                stream <<< indentStreamable(offset: 2)
+                item.write(to: stream, indent: indent.flatMap {$0 + 2})
             }
-            stream <<< "]"
+            stream <<< (shouldIndent ? "\n" : "") <<< indentStreamable() <<< "]"
```

Comment:
it indents the value passed in param. And offset is used to indent more than the passed value (indent+offset)